flybywiresim/a32nx
# fix(fuel): eliminates inner tank zeroing when re/de-fueling

Fixes #6304

## Summary of Changes

This PR fixes an issue with re-/de-fueling where, in certain cases, **inner tanks would be zeroed** while outer tanks would be draining fuel and FOB decreasing (with engines off). 

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): TazX [Z+2]#0405

## Testing instructions

1. Start the plain C&D at the Gate
2. Bring out the W&B/ Fuel UI menu from the Sim (to check tank levels)
3. Go to the Fuel tab in the EFB and **defuel completely the aircraft**
4. Enter 100Kgs of Fuel in the _input box_
    - Check outer tanks fill with 100 Kgs
    - Fuel is being transferred to inner tanks
    - FOB remains constant
6. Enter 400 Kgs of Fuel again in the _input box_
    - Check outer tanks fill with 400 Kgs
    - Fuel is being transferred to inner tanks
    - FOB remains constant

Note: before this fix, on step 6 you would see outer tanks fill to 400 Kgs, fuel on inner would be zeroed, fuel on outer would be draining and FOB would be decreasing


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
